### PR TITLE
Clear article option when all images contain alt text

### PR DESCRIPTION
**Pull request by yum25**
- Implements https://github.com/michigandaily/alt-text-tracker/issues/7

Displays a blue "clear" button when all alt text has been added to images that previously did not have alt text. When clicked, the article is refetched through the endpoint and updated to reflect its current status on the D1 database.

Things to consider before merge:
- Is there a better api implementation of this? Currently using `PUT` because it denotes idempotent behavior unlike `POST` or `PATCH`, however results in more client-side-like behavior when server-side behavior is best. (Currently force reloading the page to reflect changes to the database after the response)
- Might want to address https://github.com/michigandaily/alt-text-tracker/issues/5 in conjunction, as this allows those posts to be cleared. Do we want them to be cleared, or to alert that there's an invalid image? Could also be considered along with warning for redundant alt text language, as it would likely result in minor changes to the database
- Is refetching the article the best implementation? Could be a bit redundant, but protects against potential bugs or attempts that could clear articles that still need alt text added to them.

### CODE REVIEW

**Comment by erxclau** [comment]
A note on #5:

image

This card used to be the crossword image. It showed the crossword image but after I added alt text to the image, it shows 1/0 now, which I guess makes sense? I'm guessing the 1 comes from the database and the 0 comes from the WordPress API. So maybe #5 is about invalid images and about corrected images? I think the right thing to do here is just to remove the arrows and 1/0. Clicking the clear button does seem to work!

**Comment by erxclau** [comment]
> Is there a better api implementation of this? Currently using PUT because it denotes idempotent behavior unlike POST or PATCH, however results in more client-side-like behavior when server-side behavior is best. (Currently force reloading the page to reflect changes to the database after the response)

The "better" approach here is to use SvelteKit [form actions](https://kit.svelte.dev/docs/form-actions). It is essentially just a form, so it works without any client-side behavior. However, it can be progressively enhanced to include client behavior (like disabling the button as the form submits, or showing a popup when the form succeeds or fails). If a user does not have JavaScript enabled (or more realistically, a user is interacting with the page before JavaScript has loaded), the default web form behavior will occur (send a request, and refresh the page). Note however that HTML forms do not support PUT methods (they only support GET and POST) so you lose a bit of semantic meaning. But, I don't think it is a terrible loss.

**Comment by erxclau** [comment]
> Is refetching the article the best implementation? Could be a bit redundant, but protects against potential bugs or attempts that could clear articles that still need alt text added to them.

I think given the endpoints we have, probably the best approach.

**Comment by erxclau** [comment]
Here's an example of using server actions in the games repository:

- Form: <https://github.com/michigandaily/games/blob/374d6991cb73dcf041873a083ecaeb644d8ea1c4/src/routes/admin/edit/%5Bidentifier%5D/%2Bpage.svelte#L307-L326>
- Actions: <https://github.com/michigandaily/games/blob/374d6991cb73dcf041873a083ecaeb644d8ea1c4/src/routes/admin/edit/%5Bidentifier%5D/%2Bpage.server.ts#L137-L150>

File: src/routes/posts/+page.server.ts [hunk 1]
```diff
@@ -51,3 +55,47 @@ export const load: PageServerLoad = async ({ platform, url }) => {
...
+			.bind(images_without_alt_text, data.get('id'))
+			.run();
+
+		if (response.error) {
+			fail(400, { message: response.error });
+		}
+
+		if (response.meta.rows_written < 1) {
+			fail(400, { message: "Failed to update article"})
+		}
```
Comment (erxclau) [inline src/routes/posts/+page.server.ts:97]: I believe you need to return the failure. Personally, I think it's weird that you don't have to return for `error` or `redirect` but you have to for `fail` but that's how it was designed.

**Comment by yum25** [comment]
NOTE: didn't catch before I pushed but it's returning to the first page of /posts right now, trying to fix that :(

**Comment by erxclau** [comment]
> NOTE: didn't catch before I pushed but it's returning to the first page of /posts right now, trying to fix that :(

You _might_ not need to pass the pathname as a hidden input. The `request` parameter in your action function may be useful.

**Review by erxclau — APPROVED** [review]
cool

**Pull request merged**